### ISSUE TITLE
`spirv-link` fails when `NUMBA_DPPY_DEBUG=1`

### ISSUE THREAD

**Issue by PokhodenkoSA** [IntelPython/numba-dpex#650]
Scenario: "run dpbench gpairs workload"
```bash
NUMBA_DPPY_DEBUG=1 python run_gpairs.py --test --size 64
```
```python-traceback
...
  File "/localdisk/work/spokhode/numba-dppy/numba_dppy/spirv_generator.py", line 115, in link
    check_call(["spirv-link", *flags, "-o", opath, *binaries])
  File "/localdisk/work/spokhode/numba-dppy/numba_dppy/spirv_generator.py", line 37, in check_call
    return _real_check_call(*args, **kwargs)
  File "/localdisk/work/spokhode/miniconda3/envs/dev/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['spirv-link', '--allow-partial-linkage', '-o', '/tmp/tmpzkvy2w_v/2-linked-spirv', '/tmp/tmpzkvy2w_v/1-generated-spirv', '/localdisk/work/spokhode/numba-dppy/numba_dppy/ocl/atomics/atomic_ops.spir']' returned non-zero exit status 1.
```

**Comment by PokhodenkoSA** [comment]
@reazulhoque 
Any ideas?

**Comment by PokhodenkoSA** [comment]
It works if native atomics enabled.

**Comment by ZzEeKkAa** [comment]
Fixed in #1103

**Comment by diptorupd** [comment]
Native atomics is enabled by default and emulated atomics code is removed. The problem most likely was the `*.cl` kernel for atomics emulation was not pre-compiled with `-g` when the debug flag was used it was linking to a binary without debug symbols.

Since, emulation is now removed the problem is no longer reproducible.
